### DISCUSSION TITLE
timescale extension enabled: function time_bucket does not exist

### DISCUSSION THREAD

**Discussion by stsmedia** [supabase/supabase#23365] [Questions] [answered]
I have the timescaledb extension enabled and created a table. I am able to successfully query it directly using the time_bucket function like so:

```sql
SELECT time_bucket('5 minute', time) AS bucket,
  round( CAST(avg(price) as numeric), 2)  AS price, symbol
FROM market_data
where time > now() - '1 day'::interval 
and symbol = any (array['appl'])
GROUP BY symbol, bucket
ORDER BY bucket ASC;
```

but when I try to wrap this query in a new function like so:

```sql
create or replace function data_buckets(start_time timestamp, end_time timestamp, in_interval text, in_symbols text[])
  returns SETOF market_data
as
$$
SELECT time_bucket(in_interval, time) AS bucket,
  round( CAST(avg(price) as numeric), 2)  AS price, symbol
FROM market_data
where time >= start_time and time <= end_time 
and symbol = any (in_symbols)
GROUP BY symbol, bucket
ORDER BY bucket ASC;
$$
language sql
stable;
```

.. I get an error: 

```ERROR:  42883: function time_bucket(text, timestamp with time zone) does not exist
LINE 5: SELECT time_bucket(in_interval, time) AS bucket,
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
```

It seems like it does not look up the extensions namespace when creating the function. What can I do about that?

**Comment by GaryAustin1** [comment]
If your timescale extension was installed in extensions and you did not edit the default API search path then I would suspect your in_interval type needs to be interval type, or you need to cast it to that in the function call.   I don't use timescaledb though.

**Reply by GaryAustin1** [reply]
It is not defined as text in the function definition but as an interval.  Cast it as in interval in the call and see if that solves it.
It is not just saying it can't find the name, it is saying it can't find it with those argument types.

**Reply by GaryAustin1** [reply, marked as answer]
![image](https://github.com/supabase/supabase/assets/54564956/d5f4ab3b-9fb8-4f39-9a60-a0105697f58f)

**Comment by stsmedia** [comment]
Thanks for the response Gary. The extension was installed in extensions via the supabase web interface and I have not changed the path. The in_interval is a simple text label (could be named better) as you can see from the initial query, so that is ok I think. 

The error is indicating it cannot find the time_bucket function when creating the PG function. Wondering if I should make this a bug report?

**Comment by stsmedia** [comment]
Yes indeed, casting it to an interval did the trick. Thanks so much for your help!
